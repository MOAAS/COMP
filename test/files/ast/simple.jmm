import static io.println(int);
import ioPlus.printResult(int, boolean) void;

    /* this is a gamer comment tbh */
class test extends testerest {
    int i; int j;
    public int func(boolean a) {
        ah ah; ah ah; staying alive; staying alive;
        ah ah; ah ah; staying alive; staying aliiiiiiiiiiiiive;
        boolean a;
        int a;
        int[] b;
        c;
        d[2][4].oi(a,b,c);
        e[2] = 3;

        // c. fop-sa fl-s ur-a fi-sofc siof ipf soifu
        b = 3 / 8;
        false[true];

        element[new int[6]] = new Object();

        longaceess[4 * 2 + 1 + 9 / 8];

        this.function(this * 4, this / 3);

        while(true) oi; oi;
        if (false && 20 && (0-10+20+20))
           a[1] = 455 + 20 * 10 * 20 / a20[3];
        else true;
        return a;
    }


    public static void main(String[] args) {
          io.println(Hello);
    }
}