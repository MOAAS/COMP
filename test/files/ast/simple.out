Program
 ImportDeclaration
  ImportLifetime
  ImportName
   io
   println
  ImportArgs
   int
  ImportType
 ImportDeclaration
  ImportLifetime
  ImportName
   ioPlus
   printResult
  ImportArgs
   int
   boolean
  ImportType
   void
 ClassDeclaration
  ClassInfo
   test
   testerest
  VarDeclaration
   int
   i
  VarDeclaration
   int
   j
  RegularMethodDeclaration
   MethodType
    int
   MethodName
    func
   MethodVars
    MethodVar
     boolean
     a
   MethodContent
    VarDeclaration
     ah
     ah
    VarDeclaration
     ah
     ah
    VarDeclaration
     staying
     alive
    VarDeclaration
     staying
     alive
    VarDeclaration
     ah
     ah
    VarDeclaration
     ah
     ah
    VarDeclaration
     staying
     alive
    VarDeclaration
     staying
     aliiiiiiiiiiiiive
    VarDeclaration
     boolean
     a
    VarDeclaration
     int
     a
    VarDeclaration
     int[]
     b
    c
    Dot
     Access
      Access
       d
       2
      4
     DotArgs
      oi
      a
      b
      c
    Assignment
     Access
      e
      2
     3
    Assignment
     b
     Div
      3
      8
    Access
     False
     True
    Assignment
     Access
      element
      ArrayInit
       6
     ObjectInit
      Object
      ObjectInitArgs
    Access
     longaceess
     Add
      Add
       Mul
        4
        2
       1
      Div
       9
       8
    Dot
     This
     DotArgs
      function
      Mul
       This
       4
      Div
       This
       3
    While
     True
     oi
    oi
    If
     And
      And
       False
       20
      Add
       Add
        Sub
         0
         10
        20
       20
     Assignment
      Access
       a
       1
      Add
       455
       Div
        Mul
         Mul
          20
          10
         20
        Access
         a20
         3
     True
   Return
    a
  MainDeclaration
   ArgName
    args
   MethodContent
    Dot
     io
     DotArgs
      println
      Hello
